test/functional/collection.test.js

```diff
@@ -1307,4 +1312,34 @@ describe('Collection', function() {
       });
     }
   });
+
+  it('should createIndex with serverSelection readPreference primary', {
+    metadata: {
+      requires: { topology: 'replicaset' }
+    },
+    test: function(done) {
+      const configuration = this.configuration;
+      const client = configuration.newClient(configuration.writeConcernMax(), {
+        useUnifiedTopology: true,
+        readPreference: 'primaryPreferred'
+      });
+      client.connect((err, client) => {
+        expect(err).to.not.exist;
+        const db = client.db(configuration.db);
+        const collection = db.collection('db-two');
+        TopologyStub = sinon.spy(Topology.prototype, 'selectServer');
```

Comment:
I think a spy makes sense here because I'm not looking to change how the `selectServer` method works, simply checking to see what argument was passed into it. Is there some reason why I'd want to create & call a fake version of the `selectServer` method?

Reply:
I think a spy is fine, was just calling out that you're calling the spy a stub and what a stub would otherwise look like

Reply:
Yeah I think a spy is totally fine, I was just calling out that you're assigning a spy to a variable named "stub", and showing what a stub would otherwise look like

Reply:
Gotcha, renamed!